pallav/fashionTech/pallavBanka/parkingAssignment/Parking.java

```diff
@@ -1,2 +1,173 @@
-package com.fashionTech.pallavBanka.parkingAssignment;
+package fashionTech.pallavBanka.parkingAssignment;
 
+import java.util.Date;
+import java.util.Timer;
+
+abstract class parking{
+    void parkVehicle(vehicle v){
+        int regno;
+        short floor;
+        int slotid;
+        regno=v.regno;
+        floor=v.floor;
+        slotid=v.slotid;
+    }
+    void getVehicle(vehicle v,int regno){
+        v.regno=regno;
+        if(v.outTime != null) {
+            //difference in time in milliseconds
+            long diff = v.outTime.getTime() - v.inTime.getTime();
+            long diffHours = diff / (60 * 60 * 1000) % 24;
+            //charges Rs20/hour
+            float totalCharges = 20 * diffHours;
+            System.out.println("Your floor is:" + v.floor + "Your Slot ID is:" + v.slotid + "Your charges are:" + totalCharges);
+        }else
+            System.out.println("The vehicle hasn't left the premises.");
+    }
+}
+class vehicle extends parking{
+    int regno;
+    short floor;
+    int slotid;
+    Date inTime;
+    Date outTime;
+    void drive(){
+
+    }
+}
+class twoWheelers extends vehicle{
+    short floor;
+    /*int slotid[];
+    void slotid(){
+        for(int i=0;i<30;i++){
+            this.slotid[i]=i+1;
+        }
+    }*/
+    void floor(short floor) {
+        this.floor=floor;
+        if (floor == 1 || floor == 2 || floor==5){
+            System.out.println("You are on floor:" + floor + "This floor is for two wheeler.");
+        } else
+            System.out.println("You are on the wrong floor.");
+    }
+    void drive(){
+
+    }
+}
+class scooties extends twoWheelers{
+    int regno;
+    Date inTime;
+    Date outTime;
+    int slotid;
+    void drive(){
+        if(slotid>30) {
+            System.out.println("This floor is full");
+        }else
+            System.out.println("Your floor is:" + floor + "Your Slot Id is:" + slotid);
+    }
+}
+class cycle extends twoWheelers{
+    int regno;
+    Date inTime;
+    Date outTime;
+    int slotid;
+    void drive(){
+        if(slotid>30) {
+            System.out.println("This floor is full");
+        }else
+            System.out.println("Your floor is:" + floor + "Your Slot Id is:" + slotid);
+    }
+}
+class sportsBike extends twoWheelers{
+    int regno;
+    Date inTime;
+    Date outTime;
+    int slotid;
+    void drive(){
+        if(slotid>30) {
+            System.out.println("This floor is full");
+        }else
+            System.out.println("Your floor is:" + floor + "Your Slot Id is:" + slotid);
+    }
+}
+class cruiserBike extends twoWheelers{
```

Comment:
Each one should be a seperate class file, don't list all classes in a single file.